azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py

```diff
@@ -0,0 +1,127 @@
+                                              resolve_instance,
+                                              resolve_instance_and_project)
+
+logger = get_logger(__name__)
+
+
+def list_pools(pool_name=None, pool_type=None, action=None, properties=None, organization=None, detect=None):
+    """ (PREVIEW) List agent pools
+    :param pool_name: Filter the list with matching pool name.
+    :type pool_name: str
+    :param pool_type: Filter the list with type of pool.
+    :type pool_type: str
+    :param action: Filter the list with user action permitted.
+    :type action: str
+    :param properties: Filter the list by agent pool properties. Comma separated list.
+    :type properties: str
+    """
+    organization = resolve_instance(organization=organization, detect=detect)
+    task_agent_client = get_new_task_agent_client(organization=organization)
+    if properties:
+        properties = list(map(str, properties.split(',')))
+    return task_agent_client.get_agent_pools(
+        pool_name=pool_name, properties=properties, pool_type=pool_type, action_filter=action)
+
+
+def show_pool(pool_id, action=None, properties=None, organization=None, detect=None):
+    """ (PREVIEW) Show agent pool details
+    :param pool_id: Id of the pool to list the details.
+    :type pool_id: int
+    :param action: Filter the list with user action permitted.
+    :type action: str
+    :param properties: Filter the list by agent pool properties. Comma separated list.
+    :type properties: str
+    """
+    organization = resolve_instance(organization=organization, detect=detect)
+    task_agent_client = get_new_task_agent_client(organization=organization)
+    if properties:
+        properties = list(map(str, properties.split(',')))
+    return task_agent_client.get_agent_pool(pool_id=pool_id, properties=properties, action_filter=action)
+
+
+def list_agents(pool_id, agent_name=None, include_capabilities=None, include_assigned_request=None, properties=None,
+                include_last_completed_request=None, demands=None, organization=None, detect=None):
+    """ (PREVIEW) Get a list of agents in a pool
+    :param pool_id: The agent pool containing the agents.
+    :type pool_id: int
+    :param agent_name: Filter on agent name.
+    :type agent_name: str
+    :param include_capabilities: Whether to include the agents' capabilities in the response.
+    :type include_capabilities: bool
+    :param include_assigned_request: Whether to include details about the agents' current work.
+    :type include_assigned_request: bool
+    :param include_last_completed_request: Whether to include details about the agents' most recent completed work.
+    :type include_last_completed_request: bool
+    :param properties: Filter which custom properties will be returned. Comma separated list.
+    :type properties: str
+    :param demands: Filter by demands the agents can satisfy. Comma separated list.
```

Comment:
Example for 'demands' param as well

Reply:
Will ask the team owning the API about how the filtering with demands work.